lib/dependencies/ImportWeakDependency.js

```diff
@@ -5,24 +5,37 @@
 
 "use strict";
 
+const Dependency = require("../Dependency");
 const makeSerializable = require("../util/makeSerializable");
 const ModuleDependency = require("./ModuleDependency");
 
 /** @typedef {import("webpack-sources").ReplaceSource} ReplaceSource */
 /** @typedef {import("../Dependency")} Dependency */
 /** @typedef {import("../DependencyTemplate").DependencyTemplateContext} DependencyTemplateContext */
+/** @typedef {import("../ModuleGraph")} ModuleGraph */
 
 class ImportWeakDependency extends ModuleDependency {
-	constructor(request, range) {
+	constructor(request, range, names) {
 		super(request);
-
+		this.referencedExportNames = names;
 		this.range = range;
 		this.weak = true;
 	}
 
 	get type() {
 		return "import() weak";
 	}
+
+	/**
+	 * Returns list of exports referenced by this dependency
+	 * @param {ModuleGraph} moduleGraph module graph
+	 * @returns {string[][]} referenced exports
```

Comment:
I think the return type need to be changed to support passing `canMangle` which then sets `canMangleUsed`